packages/node-telescope-frontend/src/components/ErrorFallback.tsx

```diff
@@ -0,0 +1,37 @@
+import React from 'react';
+import { type FallbackProps } from 'react-error-boundary';
+import { Button, Col, Result, Row, Space } from 'antd';
+import { useRoutePrefix } from '../context/RoutePrefixContext';
+
+export const ErrorFallback: React.FC<FallbackProps> = ({
+  error,
+  resetErrorBoundary,
+}: FallbackProps) => {
+  const routePrefix = useRoutePrefix();
+
+  const navigateToHomePage = (): void => {
+    window.location.href = routePrefix;
```

Comment:
**ErrorBoundary** placed before <Router> provider, because of that **useLocation/useNavigate** aren't available here.

Reply:
hi @piconkayal with this approach we won't have the last routing state when we are redirected to the Error Fallback page, however we're sending the user to the dashboard again so it's okay Ig but stil UX wise it's kind off weird wdyt? . Have you checked when you redirect the page to `window.location.href = routePrefix;`  it then redirects the user to `routePrefix/requests again`?

Reply:
Thank you for the feedback @mdipanjan 

Yes, that's a major drawback when routing through **Window API**. e.g. **location.href**, **location.assign**

We can limit the error bubbling to the component level by wrapping the **Routed Component** with **ErrorFallback**. 

That gives us two benefits:

1.  Only the **Outlet** area will show the fallback component. Consumers can retry the action or navigate to different pages through the sidebar. 
2. We can use react-router-dom APIs. e.g. Link, useNavigate, useLocation, etc!

This is similar to the next.js error-handling pattern. We can still keep the global error boundary for any unexpected runtime errors, but with a button, report if the issue persists. wdyt?

Good catch on the **routePrefix** to **routePrefix/requests** transition. Since the router is configured to replace **routePrefix** with **routePrefix/requests**, we can directly go to /requests.

Reply:
Okay, Let's merge this and we can add the finer control later as an enhancement. @piconkayal